tools/graph_loading/construct_graph.py

```diff
@@ -0,0 +1,610 @@
+""" Preprocess the input data """
+from multiprocessing import Process
+import multiprocessing
+import glob
+import os
+import dgl
+import json
+import pyarrow.parquet as pq
+import pyarrow as pa
+import numpy as np
+from functools import partial
+import argparse
+
+from transformers import BertTokenizer
+import torch as th
+
+##################### The I/O functions ####################
+
+def read_data_parquet(data_file):
```

Comment:
Your question is why we need to create a function to read parquet file?
First, this is a simple wrapper function to read parquet file. Because parquet format doesn't support multi-dimension arrays, it also does some data transformation.

Reply:
We'll need to add more functions to support other formats, such as JSON and CSV.